src/F23.StringSimilarity/ShingleBased.cs

```diff
@@ -59,7 +59,7 @@ protected ShingleBased(int k)
         
         protected ShingleBased() : this(DEFAULT_K) { }
 
-        public IDictionary<string, int> GetProfile(string s)
+        public Dictionary<string, int> GetProfile(string s)
```

Comment:
Why change the public API to return the concrete type?

Reply:
Benchmarking showed an improvement by avoiding a virtual call through the interface. Really this API should probably not be public, but this is at least a largely backwards-compatible change for any users that use this method.

Reply:
This method can be made `protected internal` as it's used by the unit tests. I'll make that change!